cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -209,15 +210,21 @@ func (as *AgentPool) IncreaseSize(delta int) error {
 	}
 	ctx, cancel := getContextWithCancel()
 	defer cancel()
-	_, err = as.manager.azClient.deploymentsClient.CreateOrUpdate(ctx, as.manager.config.ResourceGroup, newDeploymentName, newDeployment)
 	klog.V(3).Infof("Waiting for deploymentsClient.CreateOrUpdate(%s, %s, %v)", as.manager.config.ResourceGroup, newDeploymentName, newDeployment)
-	if err != nil {
-		return err
+	resp, err := as.manager.azClient.deploymentsClient.CreateOrUpdate(ctx, as.manager.config.ResourceGroup, newDeploymentName, newDeployment)
+	if isSuccessHTTPResponse(resp) {
+		klog.V(3).Infof("deploymentsClient.CreateOrUpdate(%s, %s, %v) success", as.manager.config.ResourceGroup, newDeploymentName, newDeployment)
+		as.curSize = int64(expectedSize)
```

Comment:
any code comments for the these two lines of code, why we set here since it's already succeeded?
```
		as.curSize = int64(expectedSize)
		as.lastRefresh = time.Now()
```

Reply:
This is for updating cache.

Reply:
then would you add the comments?

Reply:
of course